inventory-dialog.inc

```diff
@@ -463,21 +463,47 @@ _inv_displaySlotOptions(playerid, slotid) {
 	return 0;
 }
 
-hook OnPlayerKeyStateChange(playerid, newkeys, oldkeys) {
-	if(!IsPlayerInAnyVehicle(playerid)) {
-		Logger_Dbg("inventory-dialog", "OnPlayerKeyStateChange",
-			Logger_I("playerid", playerid),
-			Logger_I("newkeys", newkeys),
-			Logger_I("oldkeys", oldkeys));
-
-		if(newkeys & INV_KEY_OPEN) {
-			DisplayPlayerInventory(playerid);
-		} else if(newkeys & INV_KEY_PUTAWAY) {
-			_inv_putAway(playerid);
+#if defined _INC_open_mp
+    hook OnPlayerKeyStateChange(playerid, KEY:newkeys, KEY:oldkeys) 
+	{
+		if (!IsPlayerInAnyVehicle(playerid))
+        {
+			Logger_Dbg("inventory-dialog", "OnPlayerKeyStateChange",
+					Logger_I("playerid", playerid),
+					Logger_I("newkeys", newkeys),
+					Logger_I("oldkeys", oldkeys));
+
+			if (newkeys & INV_KEY_OPEN) {
+				DisplayPlayerInventory(playerid);
+			} else if (newkeys & INV_KEY_PUTAWAY) {
+				_inv_putAway(playerid);
+			}
+		}
+		return 1;
+	}
+#else
+    hook OnPlayerKeyStateChange(playerid, KEY:newkeys, KEY:oldkeys)
+	{
+		if (!IsPlayerInAnyVehicle(playerid)) {
+			Logger_Dbg("inventory-dialog", "OnPlayerKeyStateChange",
+					Logger_I("playerid", playerid),
+					Logger_I("newkeys", newkeys),
+					Logger_I("oldkeys", oldkeys));
+
+			if (newkeys & INV_KEY_OPEN) {
+				DisplayPlayerInventory(playerid);
+			} else if (newkeys & INV_KEY_PUTAWAY) {
+				_inv_putAway(playerid);
+			}
 		}
+		return 1;
 	}
+#endif
 
-	return 1;
+timer PlayerPutItemInInventory[300](playerid, itemid)
```

Comment:
timer functions are generally declared after the function that initiates them (in this case, _inv_putAway)

Reply:
We actually should put timers before we use them to avoid compiler making an extra pass.
I believe it had been unnoticed for so long because in the S&S GM such parsing issues were disabled by a pragma option

Reply:
ah this makes sense, I had no idea it was ignored for so long!